modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/ServiceTemplateManage.java

```diff
@@ -345,6 +384,21 @@ public ServiceTemplateEntity getServiceTemplateDetails(UUID id, boolean checkNam
         return existingTemplate;
     }
 
+    private void updateServiceTemplateState(boolean isAutoApproveEnabled,
+                                            ServiceTemplateRequestType requestType,
+                                            ServiceTemplateEntity serviceTemplate) {
+        if (isAutoApproveEnabled) {
+            serviceTemplate.setServiceTemplateRegistrationState(
+                    ServiceTemplateRegistrationState.APPROVED);
+            serviceTemplate.setAvailableInCatalog(true);
+            serviceTemplate.setIsUpdatePending(false);
+        } else {
+            serviceTemplate.setIsUpdatePending(requestType == ServiceTemplateRequestType.UPDATE);
+            serviceTemplate.setServiceTemplateRegistrationState(
```

Comment:
the registration state must not change with update. 
The registration state is only changed and managed with first/initial registration. After that it remains untouched.